projects/packages/connection/legacy/class-jetpack-tracks-client.php

```diff
@@ -0,0 +1,230 @@
+<?php
+/**
+ * Legacy Jetpack Tracks Client
+ *
+ * @package automattic/jetpack-tracking
+ */
+
+use Automattic\Jetpack\Connection\Manager;
+
+/**
+ * Jetpack_Tracks_Client
+ *
+ * @autounit nosara tracks-client
+ *
+ * Send Tracks events on behalf of a user
+ *
+ * Example Usage:
+```php
+	require( dirname(__FILE__).'path/to/tracks/class-jetpack-tracks-client.php' );
+
+	$result = Jetpack_Tracks_Client::record_event( array(
+		'_en'        => $event_name,       // required
+		'_ui'        => $user_id,          // required unless _ul is provided
+		'_ul'        => $user_login,       // required unless _ui is provided
+
+		// Optional, but recommended
+		'_ts'        => $ts_in_ms,         // Default: now
+		'_via_ip'    => $client_ip,        // we use it for geo, etc.
+
+		// Possibly useful to set some context for the event
+		'_via_ua'    => $client_user_agent,
+		'_via_url'   => $client_url,
+		'_via_ref'   => $client_referrer,
+
+		// For user-targeted tests
+		'abtest_name'        => $abtest_name,
+		'abtest_variation'   => $abtest_variation,
+
+		// Your application-specific properties
+		'custom_property'    => $some_value,
+	) );
+
+	if ( is_wp_error( $result ) ) {
+		// Handle the error in your app
+	}
+```
+ */
+class Jetpack_Tracks_Client {
```

Comment:
@zinigor I haven't tested this, but do we need some type of checking to ensure there isn't also another plugin with an older version of Tracking already?

Thinking outloud, I'm running Jetpack latest with this new Connection package (which would load over older ones), but also running ... Backup plugin which requires an older Connection package and the Tracking package, how do we ensure these two both don't load?

Reply:
That's a great point, thank you! I'll add testing instructions for that case tomorrow, but looking at our usage of these files it looks like we're not using them outside the autoloader functionality anymore. This should make sure that we're safe. Do you think we should add a class check anyway?

Reply:
So I have tested this finally with the latest Autoloader helper, and it seems that we have had this working before in the same way:
![image](https://user-images.githubusercontent.com/374293/161124367-87af36ea-59e0-435d-9419-a9b930f1e5e4.png)

You can see here that the class has been there in the top namespace in other plugins too, including the latest Jetpack products in the wild currently. So we're not changing anything, just moving the file.